mage_ai/cache/block_action_object/__init__.py

```diff
@@ -254,13 +254,18 @@ async def initialize_cache_for_all_objects(self) -> None:
                 include_content=True,
             )
 
-        for path_dict in build_repo_path_for_all_projects(mage_projects_only=True).values():
-            repo_path = path_dict['full_path']
+        traversed_paths = dict()
+        for path_dict in build_repo_path_for_all_projects(mage_projects_only=False).values():
+            repo_path = path_dict['root_project_full_path']
 
             for block_type in BlockType:
                 file_directory_name = Block.file_directory_name(block_type)
                 directory_full_path = os.path.join(repo_path, file_directory_name)
 
+                if traversed_paths.get(directory_full_path):
```

Comment:
Keep track of the paths we already traversed so we don't traverse it multiple times.